src/client/transport/socketio.ts

```diff
@@ -97,7 +97,7 @@ export class SocketIOTransport extends Transport {
   }
 
   onChatMessage(matchID, chatMessage) {
-    this.socket.emit('chat', matchID, chatMessage);
+    this.socket.emit('chat', matchID, chatMessage, this.credentials);
```

Comment:
In the other transport methods to help catch differences between server and client code, we use types from `Master` to type the arguments to `emit`. (For example, see [`onAction`](https://github.com/boardgameio/boardgame.io/blob/8f85fb0a4f2951dd66cc3d96485b3f28909b0af7/src/client/transport/socketio.ts#L89-L97).)

It would be good to add that here too:

```suggestion
    const args: Parameters<Master['onChatMessage']> = [
      matchID,
      chatMessage,
      this.credentials,
    ];
    this.socket.emit('chat', ...args);
```

Can we also add types to the method signature itself?

```js
   onChatMessage(matchID: string, chatMessage: ChatMessage) {
```